tests/framework/testutils/log_observer.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2023 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutils
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	zapobserver "go.uber.org/zap/zaptest/observer"
+)
+
+type LogObserver struct {
```

Comment:
Please unify interface naming with `pkg/expect` as you seem to be inspired to it, but introduce your own naming scheme.

Reply:
Hi @serathius , in pkg/expect package, it provides the `Expect(s string)` to return first expected line. However, we need to get first-N lines instead of one line. For instance, in `TestV3WatchRestoreSnapshotUnsync` test case, we need to know the leader has compacted twice. It's hard to follow the existing `pkg/expect` interface pattern.